cloudinit/sources/DataSourceAzure.py

```diff
@@ -1945,6 +1950,30 @@ def load_azure_ds_dir(source_dir):
     return (md, ud, cfg, {"ovf-env.xml": contents})
 
 
+def add_dhcp_config_for_primary_and_secondary_nics_to_network_config(
```

Comment:
Why is this config only applied to NICs with `hv_netvsc` as driver?

Reply:
```
When a virtual machine (VM) is created in Azure, a synthetic network interface is created for each virtual NIC in its configuration. The synthetic interface is a VMbus device and uses the netvsc driver. Network packets that use this synthetic interface flow through the virtual switch in the Azure host and onto the datacenter's physical network.
```
```
The synthetic interface always has a name in the form eth\<n\>. Depending on the Linux distribution, the VF interface might have a name in the form eth\<n\>. Or it might have a different name in the form of enP\<n\> because of a udev rule that does renaming
```
Source: https://learn.microsoft.com/en-us/azure/virtual-network/accelerated-networking-how-it-works

Reply:
I understand that:

1. Synthetic interfaces are identifiable by the `hv_netvsc` driver.
2. Synthetic interfaces are the only affected interface kinds affected by the dns problem?

Could please add a breadcrumb pointing to this, as a comment and/or in the commit message?

Reply:
> The synthetic interface always has a name in the form eth\<n\>. 
Depending on the Linux distribution, the VF interface might have a name in the form eth\<n\>. 
Or it might have a different name in the form of enP\<n\> because of a udev rule that does renaming

Your proposed solution only matches against `eth*` names, should we also match against `enP\<n\>` ones? Or is there a reason I do not see to restrict this to `eth*`?